src/test/xmeta/test_xmeta_emit.cpp

```diff
@@ -350,23 +387,30 @@ void VerifyTypeRefName(std::string const& expectedName, std::string_view current
     REQUIRE(combineName == expectedName);
 }
 
-void VerifyReturnType(ElementType const& expectedType, std::string const& expectedTypeRef, MethodDef const& method, TypeDef const& parent)
+using value_type = std::variant<ElementType, coded_index<TypeDefOrRef>, GenericTypeIndex, GenericTypeInstSig, GenericMethodTypeIndex>;
+void VerifyTypeName(std::string const& expectedType, std::string_view currentNamespace, value_type const& type)
+{
+    if (std::holds_alternative<ElementType>(type))
+    {
+        REQUIRE(ConvertTypeToString(std::get<ElementType>(type)) == expectedType);
+    }
+    else
+    {
+        VerifyTypeRefName(expectedType, currentNamespace, std::get<coded_index<TypeDefOrRef>>(type).TypeRef());
```

Comment:
The exception thrown by `std::get` may be cryptic to some. Since this is test code, prefer an explicit `REQUIRE(std::holds_alternative<coded_index<TypeDefOrRef>>(type));